src/plugins/intel_cpu/src/nodes/strided_slice.cpp

```diff
@@ -38,8 +39,113 @@ bool StridedSlice::isSupportedOperation(const std::shared_ptr<const ov::Node>& o
     return true;
 }
 
+inline int calculate_output_shape(const int& startIdx, const int& stopIdx, const int& step, const int& len) {
+    auto start = (startIdx > len - 1) ? len - 1 : ((startIdx % len + len) % len);
+    auto stop = (stopIdx > len - 1)   ? len     : ((stopIdx % len + len) % len);
+    return (stop - start + step - 1) / step;
+}
+
+class StridedSliceShapeInfer : public ShapeInferEmptyPads {
+public:
+    StridedSliceShapeInfer(const std::shared_ptr<ov::Node>& op,
+                           const IShapeInfer::port_mask_t& port_mask,
+                           const std::set<int64_t>& begin_mask,
+                           const std::set<int64_t>& end_mask,
+                           const std::set<int64_t>& new_axis_mask,
+                           const std::set<int64_t>& shrink_axis_mask)
+    : m_op(op), m_port_mask(port_mask), m_outputShape(op->get_output_partial_shape(0).rank().get_length(), 1),
+      m_begin_mask_set(begin_mask), m_end_mask_set(end_mask), m_new_axis_mask_set(new_axis_mask), m_shrink_axis_mask_set(shrink_axis_mask) {}
+
+    std::vector<VectorDims> infer(
+        const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes,
+        const std::unordered_map<size_t, MemoryPtr>& data_dependency) override {
+        const VectorDims& shapeIn = input_shapes[0].get();
+        const VectorDims& shapeBegin = input_shapes[1].get();
+        auto beginPtr = reinterpret_cast<int *>(data_dependency.at(1)->GetPtr());
```

Comment:
> Can you give short explanation why cast to `int*` is safe here? These inputs can have any integer precision so the conversion to int is already done somewhere? As int can have different bit length so how e.g. int64_t is handled when int is less than 64-bit?

I check it that the data type is dnnl::memory::data_type::s32 while the begin/end/strides is passed in i64 data type. It is converted somewhere but I cant find this place. It is much appreciated if you know it.
To make it more clear, I replace int with int32_t.

Reply:
At the moment, due to the convert precision transformation, CPU plugin works only with s32 as a typical integer type (except of course quantization, where we use s8 and u8). But, this might be changed in near future, so it is probably better to check the type.

Reply:
> At the moment, due to the convert precision transformation, CPU plugin works only with s32 as a typical integer type (except of course quantization, where we use s8 and u8). But, this might be changed in near future, so it is probably better to check the type at least to throw if the type is unexpected.

Yes. I add a data type check for begin/end/stride here.